### ISSUE TITLE
`<picture />` elements generated by `@sveltejs/enhanced-img` cannot be styled

### ISSUE THREAD

**Issue by jack-weilage** [sveltejs/kit#11017]
### Describe the problem

Classes/styles applied to `<enhanced:img />` elements are applied to the contained `<img />` element, not the outer container. This makes grid/flex layouts hard to manage without a wrapper element.

```html
<enhanced:img class="grid-item" />
```
becomes
```html
<picture>
  <img class="grid-item" />
</picture>
```

### Describe the proposed solution

The `<picture />` element could be marked as `display: contents`, or maybe classes/styles could be applied to the `<picture />`.

The example above could become either

```html
<picture style="display:contents">
  <img class="grid-item" />
</picture>
```
or

```html
<picture class="grid-item">
  <img />
</picture>
```

### Alternatives considered

As a workaround, you can wrap a `<div />` around each `<enhanced:img />` with your grid/flex logic applied to it, but this feels wrong when `<picture />` is already wrapping your images.

### Importance

would make my life easier

**Comment by vipero07** [comment]
I may be wrong, but it seems incorrect to put classes on the `<img>` in general.

I'm using tailwindcss on my project and have the following classes `row-start-3 col-start-2 my-auto max-h-[2rem] md:max-h-[2.25rem]` which work fine with a normal image, but the enhanced image doesn't behave correctly for exactly this reason. Wrapping the enhanced image works, but it is clunky.

**Comment by jack-weilage** [comment]
Another idea: Split `<enhanced:img />` into two parts, the wrapper (`<picture />`) and the image (`<img />`).

This would lose the simplicity of just adding one element to your markup, but would make it clear where your props are applied.

**Comment by eltigerchino** [comment]
> I may be wrong, but it seems incorrect to put classes on the `<img>` in general.

I think we still want to retain being able to add classes to the <img> element so we can add styles such as [object-fit](https://developer.mozilla.org/en-US/docs/Web/CSS/object-fit)

**Comment by wentallout** [comment]
I was working a grid layout and wrapping the `enhanced:img` inside a div does work.

```
<div class="grid w-full grid-cols-12 grid-rows-1 gap-8">
		<div class="col-span-12 md:col-span-7">
			<enhanced:img class="rounded object-cover" src={hero} />
		</div>
		<div class="col-span-12 md:col-span-5">		
		</div>
	</div>
```

**Comment by Rich-Harris** [comment]
I would have thought `<picture style="display: contents">` was the solution — does anyone have a repro showing that _wouldn't_ work?

**Comment by Serator** [comment]
`display: contents` won't let you, for example, select the second `enhanced:img` element. Something like:

```
...
<enhanced:img class="image" />
...

<style>
.image:nth-child(2) {...}
</style>
```

I don't use `enhanced:img`, but I believe the modern `:has()` selector solves the problem discussed here:

```
...
<enhanced:img class="image" />
...

<style>
:has(>.image):nth-child(2) {...}
</style>
```

[Demo](https://svelte.dev/playground/hello-world?version=5.56.3#H4sIAAAAAAAAE22PPU_DMBBA_4p1UyuFmFZiMWklNgY2RoyQGx_JqY5t2deEqsp_R0nasjCd7t3Xuwt40yEoeEXnghhCclas0BKjXUMB3-Qwg_q4AJ_j1DcBKG5TLzGWuUfHEzuYjP_xOnhGzxkUVJZ6UTuT807DkEyMmDTstddc5Wj8vpJz0H5GkWo-JZzrmivqmvswdaZBDcI43ml4n29pEDnVOw0tc8xKylN0wdhyoCN1aMmUITVyyuKUyTp0XfBZbuTmIJcVX2-hCWXuGw1CLl7yz6KSlvpZrsp8dotYef1DXBZN1Zq82pez4Fp5bh_qlpxdbde3Ds2WcnTmrMTBhfr4fMMDWW6VeHrshzs7hGQxKbHtB5GDIyuahOiv9XEK42S2CEEBjD8MitMJx88C2JAbyFtQ38ZlHH8BCmq6D_EBAAA)

---

As a hack-free solution to this issue, one could consider using `img` (a single, non-nested tag) with the `srcset` attribute instead of `picture`, but I don't think that would work for all projects.

**Comment by Rich-Harris** [comment]
IIUC that wouldn't allow different formats

**Comment by Serator** [comment]
> IIUC that wouldn't allow different formats

Yes, that's right. For projects with broad browser/device support, this may not be acceptable. For everyone else, WebP is already very [well supported](https://rviscomi.github.io/timebase/#feature-webp-widely-available) (widely available since March 16, 2023). [The same](https://rviscomi.github.io/timebase/#feature-avif-widely-available) (expected to become widely available on July 25, 2026) goes for AVIF, though.

---

[:has()](https://rviscomi.github.io/timebase/#feature-has-widely-available) (expected to become widely available on June 19, 2026).

**Comment by teemingc** [comment]
I think a good solution might be to detect `data-picture-*` attributes on the enhanced img element and place them on the picture element instead of the img element:
```svelte
<enhanced:img src="..." class="img" data-picture-class="picture" />
```
should translate to
```svelte
<picture class="picture">
	<img src="..." class="img" />
</picture>
```
Not sure if adding styles to the `picture` element in a Svelte style block already works or if adding a class name is required to style them this way
